tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1218,6 +1318,32 @@ def eval_fn(m):
         filter_expr = None
     _rostopic_hz(topic, window_size=window_size, filter_expr=filter_expr)
 
+
+def _rostopic_cmd_delay(argv):
+    args = argv[2:]
+    from optparse import OptionParser
+    parser = OptionParser(usage="usage: %prog delay /topic", prog=NAME)
+    parser.add_option("-w", "--window",
+                      dest="window_size", default=-1,
+                      help="window size, in # of messages, for calculating rate", metavar="WINDOW")
+
+    (options, args) = parser.parse_args(args)
+    if len(args) == 0:
+        parser.error("topic must be specified")
+    if len(args) > 1:
+        parser.error("you may only specify one input topic")
```

Comment:
Please model the topic as an argument and let `argparse` handle the checks as well as providing documentation about it.